lib/quiz/match_the_following.dart

```diff
@@ -0,0 +1,172 @@
+import 'package:flutter/material.dart';
+
+class MatchingGame extends StatefulWidget {
+  Map<String, dynamic> gameData;
+  MatchingGame(
+      {key,
+      this.gameData = const {
+        "image":
+            "https://raw.githubusercontent.com/flutter/website/master/_includes/code/layout/lakes/images/lake.jpg",
+        "question": "what is this game about?",
+        "pairs": {
+          "1": "string1",
+          "2": "string2",
+          "3": "string3",
+          "4": "string4",
+          "5": "string5",
+          "6": "string6",
+          "7": "string7",
+          "8": "string8",
+          "9": "string9",
+          "10": "string10",
+          "11": "string11",
+          "12": "string12",
+          "13": "string13",
+          "14": "string14",
+          "15": "string15",
+          "16": "string16",
+          "17": "string17",
+          "18": "string18",
+          "19": "string19",
+          "20": "string20"
+        }
+      }})
+      : super(key: key);
+  @override
+  _MatchingGameState createState() => new _MatchingGameState();
+}
+
+class _MatchingGameState extends State<MatchingGame> {
+  Map<String, dynamic> _selected = {};
+  String _leftItemTapped;
```

Comment:
Why is this a String? Shouldnt it be a boolean?

Reply:
Sir, actually here this string is performing two roles, one is to store the left item selected by the user while matching and this I am sending and storing it inside the final map _selected that is being created at end. The second role is to decide which button to disable because obviously according to the design, once a pair got selected the buttons will become disabled. So basically I used this instead of another approach where I will be using a boolean to decide whether to disable the button or not and a string variable to store the left selected item. So, I thought it would be better if I can carry out both the requirements using only a single string variable instead of using two separate ones one boolean and another string.
Thanks.